src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,304 @@
+package teammates.ui.template;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import teammates.common.util.ActivityLogEntry;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Url;
+
+public class AdminActivityLogTableRow {
+    
+    private static final int TIME_TAKEN_WARNING_LOWER_RANGE = 10000;
+    private static final int TIME_TAKEN_WARNING_UPPER_RANGE = 20000;
+    private static final int TIME_TAKEN_DANGER_UPPER_RANGE = 60000;
+    
+    private static final String MASQUERADE_ROLE_ICON_CLASS = "glyphicon glyphicon-eye-open text-danger";
+    
+    private static final String ACTION_UNSCCUSSFUL_HIGHLIGHTER_FRONT = "<span class=\"text-danger\"><strong>";
+    private static final String ACTION_UNSCCUSSFUL_HIGHLIGHTER_BACK = "</strong><span>";
+    
+    private static final String KEYWORDS_HIGHLIGHTER_FRONT = "<mark>";
+    private static final String KEYWORDS_HIGHLIGHTER_BACK = "</mark>";
+    
+    private enum TimeTakenCssClass {
```

Comment:
I prefer if there is no CSS-related stuffs at all in back-end, otherwise it will be a headache to manage (you should've experienced what a headache it is to manage the `ActivityLogEntry`, right?). For things like this, `<c:if`s in the tag file should suffice.
If there are style-related attributes that depend on a condition, then make that condition a field of this `AdminActivityLogTableRow` class and call the field from the tag file.